### PR TITLE
chore: remove quiver + remove dependency_overrides + use latest firebase

### PR DESCRIPTION
**Pull request by rrousselGit**
## Description

As a follow-up to the release of firebase_core, cleanup the dependency_overrides/quiver/versions

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/cloud_firestore/cloud_firestore/pubspec.yaml [hunk 1]
```diff
@@ -13,9 +13,8 @@ dependencies:
   flutter:
     sdk: flutter
   meta: ^1.3.0-nullsafety.0
-  quiver: ">=3.0.0-nullsafety.2 <4.0.0"
-  firebase_core: ^0.7.0
-  firebase_core_platform_interface: ^3.0.1
+  firebase_core: ^0.7.0-nullsafety+0
```
Comment (Salakar) [inline packages/cloud_firestore/cloud_firestore/pubspec.yaml:16]:
```suggestion
  firebase_core: ^0.7.0-nullsafety.0
```

File: packages/cloud_firestore/cloud_firestore_platform_interface/pubspec.yaml [hunk 2]
```diff
@@ -12,11 +12,11 @@ dependencies:
     sdk: flutter
   meta: ^1.0.5
   collection: ^1.14.3
-  firebase_core: ^0.7.0
+  firebase_core: ^0.7.0-nullsafety+0
```
Comment (Salakar) [inline packages/cloud_firestore/cloud_firestore_platform_interface/pubspec.yaml:15]:
```suggestion
  firebase_core: ^0.7.0-nullsafety.0
```

File: packages/cloud_firestore/cloud_firestore_web/pubspec.yaml [hunk 3]
```diff
@@ -14,16 +14,17 @@ dependencies:
     sdk: flutter
   http_parser: ^3.1.3
   meta: ^1.1.7
-  firebase_core: ^0.7.0
-  firebase_core_web: ^0.2.1+3
+  firebase_core: ^0.7.0-nullsafety+0
```
Comment (Salakar) [inline packages/cloud_firestore/cloud_firestore_web/pubspec.yaml:17]:
```suggestion
  firebase_core: ^0.7.0-nullsafety.0
```

There's 29 of these in total so won't spam you with all of them 😅

Comment (rrousselGit) [reply packages/cloud_firestore/cloud_firestore_web/pubspec.yaml:17]: haha sorry!

Comment (Salakar) [reply packages/cloud_firestore/cloud_firestore_web/pubspec.yaml]: No worries 😄

**Review by Salakar — APPROVED** [review]
Other than those versions this LGTM. Goodbye quiver 🎉

**Comment by SeanZom** [comment]
Follow this [Migration Guid](https://firebase.flutter.dev/docs/migration/) to update the packages, got 
```
Because firebase_remote_config >=0.8.0-dev.0 depends on firebase_remote_config_platform_interface ^0.1.0-dev.0 which depends on firebase_core ^0.5.2, firebase_remote_config >=0.8.0-dev.0 requires firebase_core ^0.5.2.
```

When the fix will be published???

**Pull request merged**